Source/Core/InputCommon/ControllerInterface/Xlib/XInput2.cpp

```diff
@@ -74,19 +81,32 @@ void PopulateDevices(void* const hwnd)
 
   XIDeviceInfo* all_masters;
   XIDeviceInfo* current_master;
+  double scroll_increment = 1.0f;
   int num_masters;
 
   all_masters = XIQueryDevice(dpy, XIAllMasterDevices, &num_masters);
 
   for (int i = 0; i < num_masters; i++)
   {
     current_master = &all_masters[i];
+
     if (current_master->use == XIMasterPointer)
     {
+      // We need to query the master for the scroll wheel's increment, since the increment used varies
+      // depending on what input driver is being used. For example, xf86-libinput uses 120.0.
+      for (int j = 0; j < current_master->num_classes; j++)
+      {
+        if (current_master->classes[j]->type == XIScrollClass)
+        {
+          XIScrollClassInfo* scroll_event =
+              reinterpret_cast<XIScrollClassInfo*>(current_master->classes[j]);
+          scroll_increment = scroll_event->increment;
```

Comment:
Should this `break` out of the loop once it finds a value?

Reply:
yep!